extensions/src/main/java/org/pentaho/platform/web/http/api/resources/ActionResource.java

```diff
@@ -88,14 +86,14 @@ public Response invokeAction(
     @QueryParam( ActionUtil.INVOKER_ACTIONID ) String actionId,
     @QueryParam( ActionUtil.INVOKER_ACTIONCLASS ) String actionClass,
     @QueryParam( ActionUtil.INVOKER_ACTIONUSER ) String actionUser,
-    final String actionParams ) {
+    final ActionParams actionParams ) {
```

Comment:
I am not sure why you are saying it will break the listener as the listener now was changed to pass in a different 4th argument to action invoker of type ActionParams instead of String. Why is this an issue? I am not familiar with chronos testing yet but with R1 help I can verify this and see if it breaks anything.

Secondly - the serialize()/fromJson() methods they have separate concerns, the latter is related to json serialization deserialization we are talking about here. The former is related to converting to the Map required used during action invocation (maybe the name can be refactored if that is confusing to you - something like getParamsMap()  etc.). 

Thirdly why creating an instance of ActionParams is a concern? (especially when the conversion method is a black box to you). Remember that the listener is nothing but an adopter to invoke an action, so there is no concern if it "adopts" - if you want I can add another method to the resource that will just call this internally - but I don't think this is cleaner - this is a REST resource that accepts JSON and ideally the signature should look like it is now.

I understand that you guys had a lot of issues with the fact that chronos did not deal well with the json produced by ActionParams, But I am not sure it is invalid Json, maybe more "sophisticated" that requires escaping. Now, we could remove that special type information added to the json but then we introduce the risk of not being able to recreate the params on the worker now I think ...??